site/archive/slash-commands/_index.md

```diff
@@ -0,0 +1,209 @@
+date: "2017-08-19T12:01:23-04:00"
+weight: 30
+aliases:
+  - /integrate/other-integrations/slash-commands/
+---
+
+Slash commands are messages that begin with `/` and trigger an HTTP request to a web service that can in turn post one or more messages in response.
+
+Unlike outgoing webhooks, slash commands work in private channels and direct messages in addition to public channels, and can be configured to auto-complete when typing. Note that while Mattermost includes a number of built-in slash commands, this document concerns itself only with the slash commands that can be configured as custom integrations.
+
+
+## Basic usage
+
+Follow [the admin guide](https://docs.mattermost.com/developer/slash-commands.html#custom-slash-command) to create a slash command.
+
+After creating the slash command, you'll be given a `token`. __Treat this as a secret.__ Anyone who has this token will be able to mimic requests to your application.
+
+If you configured a slash command with a trigger word of `test`, a request URL of `http://example.com/slash-command` and a `POST` request method, and someone posts the message
+
+```
+/test asd
+```
+
+to your Town Square channel, your endpoint will receive:
+
+```http
+POST / HTTP/1.1
+Host: 127.0.0.1:8080
+User-Agent: mattermost-5.9.0
+Content-Length: 548
+Accept: application/json
+Authorization: Token nezum4kpu3faiec7r7c5zt6tfy
+Content-Type: application/x-www-form-urlencoded
+Accept-Encoding: gzip
+
+channel_id=i3bb9xfyqt8rtbyshmyhgsj16c&
+channel_name=town-square&
+command=%2Ftest&
+response_url=http%3A%2F%2F10.0.0.5%3A8065%2Fhooks%2Fcommands%2Fzozc1xwxybdedeyz8djwjpngny&
+team_domain=rrrr&
+team_id=tsb8crrn5tgqtedpkt81b4tcya&
+text=asd&
+token=nezum4kpu3faiec7r7c5zt6tfy&
+trigger_id=NG1kM3lyN2NqYmQxcGNyc2s0Nmo5em0xb2M6azF4NGFxZGp5MzgxM2M4NG03NzFlb2M5eG86MTU1MTIwODE5NTQyNzpNRVVDSUhSdWFrdmVGZ0RhTTd6UERoMWVEVndZK2NGbXlSYUxWQ054SVRLZGdxTWZBaUVBeGQvOU95NTFOeWxiTWVsRE1ZK0d4S2FzL2Z1TUU2Y0J1bW5JbFBCOXVEVT0%3D&
+user_id=k1x4aqdjy3813c84m771eoc9xo&
+user_name=tester
+```
+
+As evidenced by the `Content-Type`, this is a typical `form-urlencoded` payload similar to what you would receive from a `<form method="POST">` element on an HTML web page.
+
+Upon receiving one of these requests, your integration should immediately confirm that the `token` provided in the request body matches the one given to you after the slash command was created. If it does not match, it is strongly recommended that you reject the request. This ensures that the request actually came from Mattermost.
+
+Once the request is verified, you can emit a message in the channel where the slash command was invoked by responding with a 200 status code and including a response body as follows:
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+Content-Length: 59
+
+{"text": "Hello, this is a response from a slash command."}
+```
+
+Be sure to set the `Content-Type` to `application/json` otherwise your response will be displayed as plaintext.
```

Comment:
In computer science, "plaintext" is specifically unencrypted text. By not setting the content type, the payload will have no content type and just be an ASCII string, which is called "plain text".

I think we should say "plain text" because we are not talking about encrypting anything.